component/topology/syncer.go

```diff
@@ -31,13 +35,15 @@ var (
 type TopologySyncer struct {
 	do              *domain.Domain
 	topologySession *concurrency.Session
+	serverInfo      *ServerInfo
 	ctx             context.Context
 	cancel          context.CancelFunc
 }
 
 func NewTopologySyncer(do *domain.Domain) *TopologySyncer {
 	syncer := &TopologySyncer{
-		do: do,
+		do:         do,
+		serverInfo: getServerInfo(),
 	}
 	syncer.ctx, syncer.cancel = context.WithCancel(context.Background())
```

Comment:
Maybe you need to pass in the program lifecycle context in order that the syncer is cancelled when there are SIGKILL?

Reply:
It's ok here, since `syncer.Stop()` will be call in `topology.Stop()`.